nomt/src/beatree/ops/update/branch.rs

```diff
@@ -406,8 +406,7 @@ impl BranchUpdater {
 struct BranchGauge {
     first_separator: Option<Key>,
     prefix_len: usize,
-    // total separator len, not counting prefix.
-    separator_len: usize,
+    separators_len: Vec<usize>,
```

Comment:
related: https://github.com/thrumdev/nomt/pull/383/files#r1743939784

you don't need a `Vec` here.

also, small grammar nit: it's more accurate to name this field `separator_lengths` (though I expect this field will go away)